fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4

```diff
@@ -455,6 +455,7 @@ VALUES: 'VALUES';
 VARBINARY: 'VARBINARY';
 VARCHAR: 'VARCHAR';
 VARIABLES: 'VARIABLES';
+VECTOR: 'VECTOR';
```

Comment:
This should be added into nonReserve in StarRocks.g4, otherwise VECTOR will be treated as a keyword, which will cause some queries can not work.
Like if there is a column named vector in table_a;
```
select vector from table_a
```
will fail after this change.

Reply:
Done